### PR TITLE
ares: init at 126

### PR DESCRIPTION
**Pull request by Madouura**
###### Motivation for this change
Newer Higan fork.
Semi-depends on https://github.com/NixOS/nixpkgs/pull/145199/commits/86630d9a5fbcecf04178e2d1f110f075ff3f8763 from #145199

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [x] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/misc/emulators/ares/default.nix [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+    cp -r ares/{Shaders,System} $out/share/ares
+
+    runHook postInstall
+  '';
+
+  meta = with lib; {
+    homepage = "https://ares.dev";
+    description = "Open-source multi-system emulator with a focus on accuracy and preservation";
+    license = licenses.isc;
+    maintainers = with maintainers; [ Madouura ];
```
Comment (bobby285271) [inline pkgs/misc/emulators/ares/default.nix:83]: Please add yourself to `maintainers/maintainer-list.nix` as a separate commit in this PR with commit message `maintainers: add Madouura`.

File: pkgs/misc/emulators/ares/default.nix [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+    sha256 = "1bs5iklpwlkyxh6gb5r2zp9d322dl06gfjmfvxy0xflhpbmlmy0s";
+    fetchSubmodules = true;
+  };
+
+  parallel-rdp = fetchFromGitHub {
+    owner = "Themaister";
+    repo = "parallel-rdp-standalone";
+    rev = "0dcebe11ee79288441e40e145c8f340d81f52316";
+    sha256 = "1avp4wyfkhk5yfjqx5w3jbqghn2mq5la7k9248kjmnp9n9lip6w9";
+    fetchSubmodules = true;
```
Comment (SuperSandro2000) [inline pkgs/misc/emulators/ares/default.nix:32]: repo has no submodules https://github.com/Themaister/parallel-rdp-standalone

File: pkgs/misc/emulators/ares/default.nix [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+stdenv.mkDerivation rec {
+  pname = "ares";
+  version = "125.1";
+
+  src = fetchFromGitHub {
+    owner = "ares-emulator";
+    repo = "ares";
+    rev = "v${version}";
+    sha256 = "1bs5iklpwlkyxh6gb5r2zp9d322dl06gfjmfvxy0xflhpbmlmy0s";
+    fetchSubmodules = true;
```
Comment (SuperSandro2000) [inline pkgs/misc/emulators/ares/default.nix:24]: repo has no submodules https://github.com/ares-emulator/ares

**Comment by Madouura** [comment]
@SuperSandro2000 Done.
Went ahead and merged the other commit into the first to keep things clean.

**Comment by SuperSandro2000** [comment]
thanks.

@Madouura please rebase on master to solve the merge conflict.

**Comment by SuperSamus** [comment]
[ares v126 has been released.](https://ares-emulator.github.io/2021/11/28/ares-v126-released.html)

File: pkgs/misc/emulators/ares/default.nix [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+    cp -r ares/{Shaders,System} $out/share/ares
+
+    runHook postInstall
+  '';
+
+  meta = with lib; {
+    homepage = "https://ares.dev";
+    description = "Open-source multi-system emulator with a focus on accuracy and preservation";
+    license = licenses.isc;
+    maintainers = with maintainers; [ Madouura ];
```
Comment (SuperSandro2000) [inline pkgs/misc/emulators/ares/default.nix:81]: Please add yourself as a maintainer by adding an alphabetically sorted entry to `maintainers/maintainer-list.nix` and commit this change into a separate commit with the message ``maintainers: add NAME`` into this PR.

Comment (Madouura) [reply pkgs/misc/emulators/ares/default.nix:81]: This PR depends on #145199 for that.

**Comment by SuperSandro2000** [comment]
@ofborg eval

**Comment by Madouura** [comment]
@SuperSandro2000 Could you reevaluate it?
The error doesn't seem to be on the PR's side.

**Comment by SuperSandro2000** [comment]
I think you need to rebase the PR on master to fix the check.

**Pull request merged**